enhancements/workload-partitioning/pao-render.md

```diff
@@ -0,0 +1,190 @@
+
+### Implementation Details/Notes/Constraints [optional]
+
+TBD
+
+### Risks and Mitigations
+
+- PAO needs to be written in such way that the render mode and the live reconcile loop always generate identical content. This should be handled by tests and/or CI.
+
+## Design Details
+
+### Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding
+to implement the design.  For instance,
+ > 1. This requires exposing previously private resources which contain sensitive
+  information.  Can we do this?
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+- What additional testing is necessary to support managed OpenShift service-based offerings?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+This feature is supposed to be GA and supported right from release. It requires an explicit user action to be activated.
+
+The plan is to provide tests, CI and documentation in the release where this will be introduced.
+
+End-to-end tests of the Workload partitioning are covered in the high level feature description.
+
+### Upgrade / Downgrade Strategy
+
+This feature is activated locally by user and follows the lifecycle of PAO itself. There is no upgrade procedure to perform.
+
+### Version Skew Strategy
+
+How will the component handle version skew with other components?
+What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- During an upgrade, we will always have skew among components, how will this impact your work?
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI, CRI
+  or CNI may require updating that component before the kubelet.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+The idea is to find the best form of an argument why this enhancement should _not_ be implemented.
+
+## Alternatives
+
+Similar to the `Drawbacks` section the `Alternatives` section is used to
+highlight and record other possible approaches to delivering the value proposed
+by an enhancement.
```

Comment:
We did identify a couple of alternatives.

1. Add PAO to the release payload so the render step could be run during bootstrapping. This was rejected because the PAO is not useful in all, or even most, clusters.
2. Wait to enable partitioning until the PAO can be installed into the cluster on day 2. We may yet take this approach in the future, but we still consider rolling out the changes to enable partitioning on a running cluster potentially dangerous and certainly disruptive.